### ISSUE TITLE
Regression is giving nan loss

### ISSUE THREAD

**Issue by sarthakpati** [CBICA/GaNDLF#204]
**Describe the bug**
Starting a regression training job is giving nan loss.

**To Reproduce**
1. Start a regression training
2. See following warning:
```
...\venv\lib\site-packages\torch\nn\modules\loss.py:446: UserWarning: Using a target size (torch.Size([1])) that is different to the input size (torch.Size([0])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  return F.mse_loss(input, target, reduction=self.reduction)
```
3. And then this:
```
Looping over training data:  50%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                                                                                                               | 189/380 [00:09<00:06, 29.60it/s]
Half-Epoch Average Train loss :  0.0
Half-Epoch Average Train mse :  nan
```

**Expected behavior**
Nan error should not happen

**Screenshots**

**GaNDLF Version**
0.0.12

**Desktop (please complete the following information):**
N.A.

**Additional context**
Reported by Noir
